### PR TITLE
feat(interview): add multi-session dashboard mode with failover recovery

### PR DESCRIPTION
**Pull request by ReqX**
## Summary

Adds an opt-in dashboard mode to the interview module that aggregates interviews from all OpenCode sessions on the same machine. Fully backward compatible — the default behavior (per-session mode) is unchanged.

### New files

| File | Purpose |
|------|---------|
| `src/interview/dashboard.ts` | Dashboard HTTP server — state cache, session registry, file scanning, failover recovery from `.md` frontmatter |
| `src/interview/document.ts` | Extracted markdown CRUD + path utilities (previously in `service.ts`). Removes `dashboard→service` import coupling. |
| `src/interview/helpers.ts` | Shared HTTP utilities — `readJsonBody` (64KB limit), `sendJson`, `sendHtml`, `isValidId`, `extractResumeSlug` |
| `src/interview/dashboard.test.ts` | 51 tests covering all dashboard HTTP routes, auth, state push/merge, answer relay, nudge |
| `src/interview/manager.test.ts` | 14 tests covering per-session vs dashboard bifurcation, session registration, state push wiring |
| `src/interview/helpers.test.ts` | 21 tests for helpers including 64KB body limit |
| `src/interview/parser.test.ts` | 27 tests for parser + Zod validation |

### Modified files

| File | Changes |
|------|---------|
| `manager.ts` | Composition root — routes to per-session or dashboard based on config. Dashboard init with failover. Session registration. Answer/nudge polling. |
| `service.ts` | Extracted document functions to `document.ts`. Added state push callback, interview created callback, nudge action handler for dashboard mode. |
| `server.ts` | Added nudge endpoint, improved answer parsing |
| `parser.ts` | Zod validation for raw LLM output (`RawInterviewStateSchema`, `RawQuestionSchema`) before normalization |
| `types.ts` | Zod schemas, `InterviewStateEntry` type, file item types |
| `ui.ts` | Dashboard page render, interview page with disconnect detection, nudge buttons, copy button fix (`stopPropagation`), download button, settings panel |
| `config/schema.ts` | Added `dashboard` boolean to interview config schema |

## Architecture

**Sessions are smart, dashboard is dumb.** Sessions drive LLM interaction locally (parse state, inject prompts, write `.md`). Dashboard is a stateless aggregator with a web UI. Zero cross-process SDK dependency.

State flows upward — sessions POST parsed state to dashboard. First process wins with auto-failover. Answer delivery via session polling.

## Configuration

| `port` | `dashboard` | Mode |
|--------|-------------|------|
| `0` (default) | `false` (default) | Per-session — each process runs its own server |
| `0` | `true` | Dashboard on default port 43211 |
| `> 0` | any | Dashboard on the specified port |

## Security

- Binds to `127.0.0.1` only
- Token auth on mutation endpoints (startup-generated, HttpOnly cookie for browser)
- PID-based stale auth file detection
- No CORS headers (same-origin only)
- Zod validation for untrusted LLM output
- ID sanitization against path traversal

## Testing

953 tests across 55 files, all passing. 113 new tests for dashboard features.

## Known limitations

- Answer delivery has a few seconds latency (polling-based)
- Browser updates use polling, not realtime push
- Runtime state is in-memory; `.md` files are the durable artifact

## Call for testers 🧪

This feature needs eager and brave testers who aren't afraid of a dashboard that might occasionally look at them funny. Multi-process coordination is inherently chaotic — your OpenCode sessions will fight over who gets to be the dashboard, interviews will reconnect from the dead, and `.md` files will be scanned from places you forgot existed.

If you enjoy finding edge cases more than you enjoy things working perfectly on the first try, this one's for you. Set `"dashboard": true` in your interview config, open a few OpenCode sessions, and let us know what breaks.

Spoiler: something will break. That's the point.

### CODE REVIEW

File: src/interview/dashboard.ts [hunk 1]
```diff
@@ -0,0 +1,1117 @@
...
+      if (!sessionList) return;
+
+      const cutoff =
+        scanDays > 0 ? Date.now() - scanDays * 24 * 60 * 60 * 1000 : 0;
+
+      for (const session of sessionList) {
+        if (!session.directory) continue;
+        if (cutoff > 0 && session.time?.updated) {
+          if (session.time.updated < cutoff) continue;
+        }
```
[the remark this answers is not in the document]
Comment (ReqX) [reply src/interview/dashboard.ts:289]: false pos, confirmed via SDK milliseconds since epoch, comparison correct

File: src/interview/dashboard.ts [hunk 2]
```diff
@@ -0,0 +1,1117 @@
...
+      const message = error instanceof Error ? error.message : String(error);
+      if (message.includes('already in use')) {
+        // Another process won the race, wait with jitter and retry
+        if (attempt < maxAttempts - 1) {
+          await new Promise((resolve) => setTimeout(resolve, jitterMs()));
+          continue;
+        }
+      }
+      throw error;
+    }
```
[the remark this answers is not in the document]
Comment (ReqX) [reply src/interview/dashboard.ts:1189]: legit, will fix

**Comment by ReqX** [comment]
@greptile-apps[bot] thanks for the review. All findings addressed in `b6d7550`:

- **P1 (EADDRINUSE on last retry):** Fixed — `return null` on exhausted retries instead of throwing. Graceful fallback to session mode.
- **P2 (console.log instead of log utility):** Fixed — replaced with shared `log()` from `../utils`, removed ESLint disable comment.
- **P2 (duplicate decodeURIComponent):** Fixed — extracted to `rawId` variable.
- **P2 (timestamp unit mismatch):** False positive — confirmed via SDK types that `session.time.updated` is milliseconds since epoch (`Filter sessions updated on or after this timestamp (milliseconds since epoch)`). Comparison with `Date.now()` is correct as-is.

**Comment by ReqX** [comment]
## Update: Post-Merge Fixes

Merged master (including #304 node runtime compat, one test conflict resolved).

Additional fixes since initial review:

- **Duplicate interview entries** — `recovered-*` entries from `rebuildFromFiles` survived until first state push (~10s). Added `dedupRecovered()` called at HTTP create, HTTP state-push, and in-process `pushState` — zero-window dedup.
- **Nudge race** — `getInterviewState` ran before nudge/answer processing, pushing `completed` over `awaiting-agent`. Reordered in `manager.ts`.
- **Banner false-positive** — Health endpoint `Date.now()` triggered "Dashboard updated" every poll. Replaced with stable `sig`.
- **UI: directory labels, sessionID-based file dedup**

955 tests pass, lint + typecheck clean.

@greptile-apps[bot] — requesting re-review, especially:
- `dedupRecovered` / `hasLiveForSlug` helpers in `dashboard.ts` (correctness + edge cases)
- Nudge reorder in `manager.ts` (~L370-415)
